backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -602,7 +621,7 @@ object Compiler {
           cause match {
             case f: StopPipelining => Result.Blocked(f.failedProjectNames)
             case f: xsbti.CompileFailed =>
-              // We cannot assert reporter.problems == f.problems, so we aggregate them together
+              // We cannot guarantee reporter.problems == f.problems, so we aggregate them together
```

Comment:
sidenote: could use an explanation why we cannot

Reply:
IIRC there were some cases when the compiler could return different problems in these two branches, hence the fix. Didn't elaborate when I introduced the fix because the root of the bug was not clear to me. I prefer to leave it as is for the moment